### PR TITLE
chore: Replace eventBus with mitt.js [CW-3275]

### PR DESCRIPTION
**Pull request by fayazara**
# Replace the deprecated `eventBus` with mitt.js

## Description

Since eventBus and it's respective methods are deprecated and removed from all future releases of vue, this was blocking us from migrating. This PR replaces eventBus with [mitt](https://github.com/developit/mitt). I have created a wrapper mitt.js to simulate the same old event names so it's backwards compatible, without making a lot of changes.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

1. Made sure all the places we're listening to bus events are working as expected.
2. Respective specsf or the events from mitt.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/packs/mitt.js
Comment (scmmishra) [inline app/javascript/packs/mitt.js:1]: Let's move this to the `dashboard/helper/` folder, similar to `createAxios`

**Review by scmmishra — CHANGES_REQUESTED** [review]

File: app/javascript/packs/mitt.js
Comment (fayazara) [reply app/javascript/packs/mitt.js:1]: Fixed in https://github.com/chatwoot/chatwoot/pull/9455/commits/218b672498b32dd7d7945aae6b34ccffca1369c9

**Comment by pranavrajs** [comment]
@fayazara A couple of thoughts. 

1/ Can we remove the usage of window.bus from the code? I don't think it is good to use the same interface anymore.
2/ Please check the widget side code as well. It is not updated there.

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Please check the comment above.

**Pull request closed without merging**